testdata/project-v2-addon/go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/kubebuilder/testdata/project-v2-addon
 
-go 1.13
+go 1.17
```

Comment:
@varshaprasad96 that was what I was looking for to know.
If we had any reason to not update the go/v2 or if we only forgot this one.

Really thank you to bring that. 
So, clearly we need to bump the go version with the k8s deps as well to avoid issues. 

In this way, I am closing this one as deferred.